### PR TITLE
support cpp17[json]

### PR DESCRIPTION
**Pull request by bbbgan**

### CODE REVIEW

**Comment by qicosmos** [comment]
do you compare the performance between version c++20 and version c++17?

**Comment by bbbgan** [comment]
>Almost the same, at least during Ubuntu and Clang testing

**Comment by bbbgan** [comment]
Almost the same, at least during Ubuntu with gcc testing

**Comment by bbbgan** [comment]
left：C++20  right： C++17
Ubuntu22.04 GCC11.3.0
image

**Comment by bbbgan** [comment]
> left：C++20 right： C++17 Ubuntu22.04 GCC11.3.0 image

The above are two versions of the current PR，The following is the benchmark of the current repository 
image

File: iguana/define.h [hunk 1]
```diff
@@ -10,3 +17,87 @@
...
+  // x will never be zero in iguana
+  return __builtin_ctzll(x);
+}
+namespace std {
+template <typename T>
+using remove_cvref_t =
+    typename remove_cv<typename remove_reference<T>::type>::type;
+}
+template <typename T>
+constexpr inline bool contiguous_iterator =
```
Comment (qicosmos) [inline iguana/define.h:88]: is this contiguous_iterator is same with above contiguous_iterator?

**Review by qicosmos — APPROVED** [review]
LGTM

**Comment by qicosmos** [comment]
improve as_likely, please look at yalantinglibs as_likely

**Pull request merged**